### DISCUSSION TITLE
Include SHA in "Download ZIP" archive

### DISCUSSION THREAD

**Discussion by brechtm** [community/community#41009] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

When downloading a ZIP for a specific branch (e.g. main) of a repository at several points in time will all yield the same filename _\<repository\>-\<branch\>.zip_. At this moment, it is not possible to figure out to which commit the ZIP archives correspond to. Therefore, I propose adding an extra file (e.g. _.commit_) to the ZIP that contains the SHA of the commit it corresponds to.

Let me also add a specific use case for this. Users of my Python project can easily install the latest development version by passing the ZIP URL to pip (the Python package installer). Unfortunately, since this information is not present in the ZIP archive, I cannot set a specific version number identifying the commit. This can cause confusion as it's not easy to tell which version someone has installed.

(It's also posssible to pass the git repo URL to pip, so that the exact version can be retrieved from git. However, that requires git to be installed and the download size for the repo is much larger than the ZIP of a single revision)

Alternatively, the (short) SHA could be included in the ZIP filename, e.g. _\<repository\>-\<branch\>-\<sha\>.zip_. This will not help the use case described above, since the installation code of a Python package will not be aware of the filename of the ZIP file. So perhaps it's possible to change the filename as well as include a _.commit_ file.

**Comment by ossanna16** [comment]
Hi there @brechtm and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
